astropy/visualization/wcsaxes/tests/test_formatter_locator.py

```diff
@@ -236,205 +288,359 @@ def test_values_unit(self):
+            ),
+            (u.degree, True, False, 15 * u.degree, '15', '15'),
+            (u.degree, True, False, 0.12 * u.degree, '15.39', '15.39'),
+            (u.degree, True, False, 0.0036 * u.arcsec, '15.392231', '15.392231'),
+            (u.arcmin, True, False, 15 * u.degree, '924', '924'),
+            (u.arcmin, True, False, 0.12 * u.degree, '923.5', '923.5'),
+            (u.arcmin, True, False, 0.1 * u.arcmin, '923.5', '923.5'),
+            (u.arcmin, True, False, 0.0002 * u.arcmin, '923.5339', '923.5339'),
+            (u.arcsec, True, False, 0.01 * u.arcsec, '55412.03', '55412.03'),
+            (u.arcsec, True, False, 0.001 * u.arcsec, '55412.032', '55412.032'),
+            (u.mas, True, False, 0.001 * u.arcsec, '55412032', '55412032'),
+            # Make sure that specifying None defaults to
+            # decimal for non-degree or non-hour angles
+            (
+                u.arcsec,
+                None,
+                True,
+                0.01 * u.arcsec,
+                '55412.03"',
+                r'$55412.03\mathrm{^{\prime\prime}}$',
+            ),
+        ],
+    )
+    def test_formatter_no_format_with_units(
+        self, format_unit, decimal, show_decimal_unit, spacing, ascii, latex
+    ):
         # Check the formatter works when specifying the default units and
         # decimal behavior to use.
-        fl = AngleFormatterLocator(unit=u.degree, format_unit=format_unit, decimal=decimal, show_decimal_unit=show_decimal_unit)
+        fl = AngleFormatterLocator(
+            unit=u.degree,
+            format_unit=format_unit,
+            decimal=decimal,
+            show_decimal_unit=show_decimal_unit,
+        )
         assert fl.formatter([15.392231] * u.degree, spacing, format='ascii')[0] == ascii
         assert fl.formatter([15.392231] * u.degree, spacing, format='latex')[0] == latex
 
     def test_incompatible_unit_decimal(self):
         with pytest.raises(UnitsError) as exc:
             AngleFormatterLocator(unit=u.arcmin, decimal=False)
-        assert exc.value.args[0] == 'Units should be degrees or hours when using non-decimal (sexagesimal) mode'
+        assert (
+            exc.value.args[0]
+            == 'Units should be degrees or hours when using non-decimal (sexagesimal)'
+            ' mode'
+        )
 
 
 class TestScalarFormatterLocator:
-
     def test_no_options(self):
-
         fl = ScalarFormatterLocator(unit=u.m)
         assert fl.values is None
         assert fl.number == 5
         assert fl.spacing is None
 
     def test_too_many_options(self):
-
         with pytest.raises(ValueError) as exc:
-            ScalarFormatterLocator(values=[1., 2.] * u.m, number=5)
-        assert exc.value.args[0] == "At most one of values/number/spacing can be specified"
+            ScalarFormatterLocator(values=[1.0, 2.0] * u.m, number=5)
+        assert (
+            exc.value.args[0] == "At most one of values/number/spacing can be specified"
```

Comment:
As above we should use match= (I know this is not strictly related to the formatting changes but I'm looking at this code for the first time in a while! Same comment below several times but I won't keep repeating.

Reply:
See  5dded7a.